ui/src/data_explorer/containers/DataExplorer.js

```diff
@@ -35,6 +38,25 @@ class DataExplorer extends Component {
     return queryConfigs[0]
   }
 
+  componentDidMount() {
+    const {query} = queryString.parse(window.location.search)
+    if (query && query.length) {
+      const qc = this.props.queryConfigs[0]
+      this.props.queryConfigActions.editRawText(qc.id, query)
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {router} = this.props
+    const {queryConfigs, timeRange} = nextProps
+    const query = buildRawText(queryConfigs[0], timeRange)
+    const qsCurrent = queryString.parse(router.location.search)
+    if (query.length && qsCurrent.query !== query) {
+      const qsNew = queryString.stringify({query})
+      router.push(`${window.location.pathname}?${qsNew}`)
+    }
```

Comment:
`router.location.pathname` should work here as well

Reply:
yep!